File: app/src/main/java/com/example/tribecrypto/data/network_object/CryptoCurrencyObject.java
package com.example.tribecrypto.data.network_object;

import com.google.gson.annotations.SerializedName;

public class CryptoCurrencyObject {
    
    @SerializedName("name") 
    private String name;
    
    @SerializedName("symbol")
    private String symbol;
    
    @SerializedName("slug")
    private String slug;

    @SerializedName("circulating_supply")
    private String circulatingSupply;

    @SerializedName("cmc_rank")
    private Long cmcRank;

    @SerializedName("date_added")
    private String dateAdded;

    @SerializedName("last_updated")
    private String lastUpdated;

    @SerializedName("quote")
    private CryptoCurrencyQuote cryptoQuote;

    public String getName() {
        return name;
    }

    public String getSymbol() {
        return symbol;
    }

    public String getSlug() {
        return slug;
    }

    public String getCirculatingSupply() {
        return circulatingSupply;
    }

    public Long getCmcRank() {
        return cmcRank;
    }

    public String getDateAdded() {
        return dateAdded;
    }

    public String getLastUpdated() {
        return lastUpdated;
    }

    public CryptoCurrencyQuote getCryptoQuote() {
        return cryptoQuote;
    }
}

File: app/src/main/java/com/example/tribecrypto/data/network_object/CryptoCurrencyObject.kt
package com.example.tribecrypto.data.network_object

import com.google.gson.annotations.SerializedName

class CryptoCurrencyObject(
@SerializedName("name") val name: String,
@SerializedName("symbol") val symbol: String,
@SerializedName("slug") val slug: String,
@SerializedName("circulating_supply") val circulatingSupply: String,
@SerializedName("cmc_rank") val cmcRank: Long,
@SerializedName("date_added") val dateAdded: String,
@SerializedName("last_updated") val lastUpdated: String,
@SerializedName("quote") val cryptoQuote : CryptoCurrencyQuote
)